yktrace/src/tir.rs

```diff
@@ -98,7 +99,8 @@ pub struct SirMarkers {
 /// A TIR trace is conceptually a straight-line path through the SIR with guarded speculation.
 #[derive(Debug)]
 pub struct TirTrace {
-    ops: Vec<TirOp>
+    ops: Vec<TirOp>,
+    trace_inputs_local: Option<Local>
```

Comment:
Did we want to kill `Option`?

Reply:
I'll double check this.

Reply:
I think in practise we should always have trace inputs (esp. since users will only use the merge point function, which does all of this behind the scenes). Though if we remove this option we should produce a proper error somewhere if trace inputs aren't set. So we should probably do this in a separate PR.

Reply:
Agreed on all counts.